src/System.Management.Automation/namespaces/EnvironmentProvider.cs

```diff
@@ -198,15 +198,33 @@ internal override IDictionary GetSessionStateTable()
             IDictionary environmentTable = Environment.GetEnvironmentVariables();
             foreach (DictionaryEntry entry in environmentTable)
             {
-                // Windows only: duplicate key (variable name that differs only in case)
-                // NOTE: Even though this shouldn't happen, it can, e.g. when npm
-                //       creates duplicate environment variables that differ only in case -
-                //       see https://github.com/PowerShell/PowerShell/issues/6305.
-                //       However, because retrieval *by name* later is invariably
-                //       case-Insensitive, in effect only a *single* variable exists.
-                //       We simply ask Environment.GetEnvironmentVariable() which value is
-                //       the effective one, and use that.
-                providerTable.TryAdd((string)entry.Key, entry);
+                try
+                {
+                    providerTable.Add((string)entry.Key, entry);
+                }
+                catch (System.ArgumentException)
```

Comment:
We could use
```c#
if (!providerTable.Add((string)entry.Key, entry))
{
    string effectiveValue = Environment.GetEnvironmentVariable((string)entry.Key); 
   ...
}

Reply:
@iSazonov: Assuming you meant `.TryAdd()`: good idea; done.